File: plugins/logger.py
from lattice import Capability, ModuleSpec


def start_logger(ctx):
    class LoggerV1:
        def info(self, msg):
            print(f"[info] {msg}")

        def warn(self, msg):
            print(f"[warn] {msg}")

        def error(self, msg):
            print(f"[error] {msg}")

    ctx.bind(Capability("logger", (1, 0)), LoggerV1())


REGISTER_MODULES = [
    ModuleSpec(
        name="logger",
        provides=[Capability("logger", (1, 0))],
        requires=[],
        start=start_logger,
    )
]

REGISTER_ADAPTERS = []

File: typescript/src/plugins/logger.ts
import type { ModuleSpec } from "../core.js";

export const REGISTER_MODULES: ModuleSpec[] = [
  {
    name: "logger",
    provides: [{ name: "logger", version: [1, 0] }],
    requires: [],
    start(ctx) {
      ctx.bind({ name: "logger", version: [1, 0] }, {
        info(message: string) {
          console.log(`[info] ${message}`);
        },
        warn(message: string) {
          console.log(`[warn] ${message}`);
        },
        error(message: string) {
          console.log(`[error] ${message}`);
        },
      });
    },
  },
];

export const REGISTER_ADAPTERS = [];
